db/migrate/20250120134632_add_unique_constraint_to_email.rb

```diff
@@ -0,0 +1,6 @@
+class AddUniqueConstraintToEmail < ActiveRecord::Migration[8.0]
+  def change
+    add_index :ect_at_school_periods, :email, unique: true, where: "email IS NOT NULL", name: "unique_email_on_ect_at_school_periods"
+    add_index :mentor_at_school_periods, :email, unique: true, where: "email IS NOT NULL", name: "unique_email_on_mentor_at_school_periods"
```

Comment:
I don't think we need the `where:` clause here, nulls aren't considered equal.

Reply:
Depends possibly - the default behaviour of the unique index is to disregard null values when enforcing uniqueness but if we are potentially going to have _lots_ of emails with 'null' then adding the WHERE clause will make the index smaller. 

Happy to take you recommendation on this!

Reply:
Will this work for teachers:
1. using their personal email and they move school and want to use the same email
2. leave a school and then return at some point in the future and wish to keep their original school email

Reply:
As discussed we are leaving constraints alone for now